core/eolearn/tests/test_utils/test_fs.py

```diff
@@ -66,6 +71,20 @@ def test_s3_filesystem(aws_session_token):
         assert filesystem.aws_session_token == aws_session_token
 
 
+@pytest.mark.parametrize("s3fs_function", [get_filesystem, load_s3_filesystem])
+def test_s3fs_keyword_arguments(s3fs_function):
+    filesystem = s3fs_function("s3://dummy-bucket/", acl="bucket-owner-full-control")
```

Comment:
Interesting, I didn't know the S3FS already provides an `acl` argument. Perhaps you could also add a test where the `acl` is set via the `upload_args`? Or is that outside of this scope and is the responsibility of S3FS?

Reply:
Actually, from the [`S3FS` implementation](https://github.com/PyFilesystem/s3fs/blob/v1.1.1/fs_s3fs/_s3fs.py#L302) you can see it is the other way around. Parameter `acl` is just a helper parameter and its value gets put into `upload_args` dictionary. This unit tests checks this.